File: Third Course/3_task.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

public partial class FormComponent : Form
{
    public FormComponent()
    {
        InitializeComponent();
    }

    private void button1_Click(object sender, EventArgs e)
    {
        Sentence go = new Sentence();
        go.countLetters();
        textBox1.Text = go.countLetters().ToString();
        go.countWords();
        textBox2.Text += go.countWords().ToString();
        go.sentenceEquality();
        textBox3.Text += go.sentenceEquality();
    }

    private void textBox1_TextChanged(object sender, EventArgs e)
    {

    }

    private void textBox2_TextChanged(object sender, EventArgs e)
    {
    }

    private void textBox1_TextChanged_1(object sender, EventArgs e)
    {

    }

    private void label1_Click(object sender, EventArgs e)
    {

    }
}


public class Sentence : FormComponent
{
    public string sentence;
    public string sentence_2;
    public Sentence() { sentence = "Hello World"; sentence_2 = "No Hello"; }
    public Sentence(string STR) { sentence = STR; sentence_2 = STR; }
    public int countLetters()
    {
        int count_letters = 0;
        for (int i = 0; i < sentence.Length; i++)
        {
            if (sentence[i] != ' ')
                count_letters++;
        }
        return count_letters;
    }


    public int countWords()
    {
        int count_words = 1;
        for (int i = 0; i < sentence.Length; i++)
        {
            if (sentence[i] == ' ')
                count_words++;
        }
        return count_words;
    }


    public string sentenceEquality()
    {
        if (sentence.Length != sentence_2.Length)
        {
            return ("Предложения не равны!");
        }
        else
            for (int i = 0; i < sentence.Length; i++)
            {
                if (sentence[i] != sentence_2[i])
                {
                    return ("Предложения не равны!");
                }
                if (i + 1 == sentence.Length)
                {
                    return ("Предложения равны!");
                }
            }
        return sentence_2;
    }
}

File: Third Course/3_task.cpp
#include "string.h"
#include "iostream"
#include <sstream>
#include <vector>
#include <array>

using namespace std;

class sentence
{
private:
    string sentence;

public:
    Sentence() { sentence = "Hello, world"; }
    Sentence(string STR) { sentence = STR; }

    void countLetters()
    {
        int count_letters = 0;
        for (int i = 0; i < sentence.length(); i++)
        {
            if (sentence[i] != ' ')
                count_letters++;
        }
        cout << "Количество букв: " << count_letters << "\n";
    }

    void countWords()
    {
        int count_words = 1;
        for (int i = 0; i < sentence.length(); i++)
        {
            if (sentence[i] == ' ')
                count_words++;
        }
        cout << "Количество слов: " << count_words << "\n";
    }

    void searchWord()
    {
        string word_to_find;
        int check_if_end = 0;
        cout << "Введите слово для поиска ";
        getline(cin, word_to_find);
        int word_to_find_len = word_to_find.length();
        int orig_word_len = sentence.length();
        cout << "Слово которое ищем: " << word_to_find << endl;
        for (int i = 0, j = 0; i < orig_word_len - 1, j < word_to_find.length(); i++)
        {
            if (sentence[i] == word_to_find[j])
            {
                check_if_end++;
                j++;
            }
            else
            {
                j = 0;
                check_if_end = 0;
            }
            if ((check_if_end == word_to_find_len) && (sentence[i + 1] == ' ') || (sentence[i + 1] == '\0') && ((check_if_end == word_to_find_len)))
            {

                cout << "Слово присутствует!";
                break;
            }
            if (i + 1 == '\0' || j + 1 > word_to_find.length())
                cout << "Слово отсутствует!";
            if (i == orig_word_len - 1)
            {
                cout << "Слово отсутствует!";
                break;
            }
        }
    }

    void sentenceEquality()
    {
        string sentence_2;
        cout << "\nВведите предложение ";
        getline(cin, sentence_2);
        if (sentence.length() != sentence_2.length())
        {
            cout << "Предложения не равны!";
            return;
        }
        else
            for (int i = 0; i < sentence.length(); i++)
            {
                if (sentence[i] != sentence_2[i])
                {
                    cout << "Предложения не равны!";
                    break;
                }
                if (i + 1 == sentence.length())
                {
                    cout << "Предложения равны!";
                }
            }
    }
};

void main()
{
    setlocale(LC_ALL, "RUS");
    string sentence;
    cout << "Введите предложение ";
    getline(cin, sentence);
    Sentence Pr;
    Sentence Pr2(sentence);
    Pr2.countLetters();
    Pr2.countWords();
    Pr2.searchWord();
    Pr2.sentenceEquality();
}
